pkg/state/common_test.go

```diff
@@ -385,3 +385,17 @@ func runFuncForObjectInSlice(objects []*unstructured.Unstructured, objectKind st
 	}
 	return found
 }
+
+// assertUnstructuredListHasKinds will check that there is an object in the list for each Kind in `kinds`.
+// kinds may have duplicates and the length of kinds must exactly match the length of objects.
+// If multiple objects have the same kind it should be in the `kinds` list multiple times.
```

Comment:
will this behave as the the description?

say:

`assertUnstructuredListHasExactKinds( [roleObj, daemonsetObj] , ["role", "role"])`

the loop at L398 will always hit the map generated for the first obj and not fail.

Reply:
You're right for sure - it's not trivial to get rid of the map here though. I'll take some other approach

Reply:
Should be better now - and I think it's a lot simpler to reason about.

Reply:
much better, thx ! :)